pkg/kv/kvserver/scheduler.go

```diff
@@ -297,7 +301,7 @@ func (s *raftScheduler) worker(ctx context.Context) {
 				state.flags |= stateRaftReady
 			}
 		}
-		if state.flags&stateRaftTick != 0 {
+		for t := state.ticks; t > 0; t-- {
 			// processRaftTick returns true if the range should perform ready
 			// processing. Do not reorder this below the call to processReady.
 			if s.processor.processTick(ctx, id) {
```

Comment:
The call stack here is:

1. `Store.processTick`
2. `Replica.tick`
3. `raft.RawNode.Tick`

We can loop over one of them (currently 1-st). Seemingly, it's more efficient to do deeper in the stack, but I wonder which one is correct. Perhaps we can refactor to enable looping over `RawNode.Tick` directly.

Reply:
Yeah, I'd loop over `processTick`. There may be logic that high up in the stack that relies (or will rely on) the tick count for correctness.